sos/policies/distros/__init__.py

```diff
@@ -296,10 +297,21 @@ def pre_work(self):
         # this method will be called before the gathering begins
 
         cmdline_opts = self.commons['cmdlineopts']
+        caseid = cmdline_opts.case_id if cmdline_opts.case_id else ""
 
         if cmdline_opts.low_priority:
             self._configure_low_priority()
 
+        # set or query for case id
+        if not cmdline_opts.batch and not \
+                cmdline_opts.quiet:
+            if caseid:
+                self.commons['cmdlineopts'].case_id = caseid
+            else:
+                self.commons['cmdlineopts'].case_id = input(
+                    _("Optionally, please enter the case id that you are "
+                      "generating this report for [%s]: ") % caseid
+                )
```

Comment:
We have the same code in `prompt_for_case_id` method in https://github.com/sosreport/sos/blob/main/sos/upload/targets/redhat.py#L55-L72 . Shouldnt we have just one routine for that identical code block?

(wont this code block here make the `prompt_for_case_id` method redundant?)